File: src/main/java/org/improving/model/Sermon.java
package org.improving.model;

import lombok.Data;

@Data
public class Sermon {
    private String date;
    private String title;
    private String mp3;
    private String preacher;
    private YouTubeInfo youTubeInfo;

    public Sermon(String date, String title, String mp3, String preacher) {
        this.date = date;
        this.title = title;
        this.mp3 = mp3;
        this.preacher = preacher;
    }

    public String toString() {
        return String.format("%s by %s (%s)", title, preacher, date);
    }
}

File: webapp/src/app/common/sermon.ts
export class Sermon {
  id: string;
  date: string;
  title: string;
  mp3: string;
  preacher: string;
  youTubeInfo: string;

  constructor(id: string, date: string, title: string, mp3: string, preacher: string, youTubeInfo: string) {
    this.id = id;
    this.date = date;
    this.title = title;
    this.mp3 = mp3;
    this.preacher = preacher;
    this.youTubeInfo = youTubeInfo;
  }

}
